internal/mesh/internal/mappers/sidecarproxymapper/computed_routes_mapper.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package sidecarproxymapper
+
+import (
+	"context"
+
+	"github.com/hashicorp/consul/internal/catalog"
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/internal/storage"
+	pbmesh "github.com/hashicorp/consul/proto-public/pbmesh/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+func (m *Mapper) MapComputedRoutesToProxyStateTemplate(ctx context.Context, rt controller.Runtime, res *pbresource.Resource) ([]controller.Request, error) {
+	var computedRoutes pbmesh.ComputedRoutes
+	err := res.Data.UnmarshalTo(&computedRoutes)
+	if err != nil {
+		return nil, err
+	}
+
+	// Each Destination gets a single ComputedRoutes.
+	serviceID := resource.ReplaceType(catalog.ServiceType, res.Id)
+	serviceRef := resource.Reference(serviceID, "")
+
+	var result []controller.Request
+
+	for port := range computedRoutes.PortedConfigs {
+		dest, ok := m.destinationsCache.ReadDestination(serviceRef, port)
+		if !ok {
+			continue // skip
+		}
+
+		for rk := range dest.SourceProxies {
+			result = append(result, controller.Request{ID: rk.ToID()})
+		}
+	}
+
+	// todo (ishustava): this is a stub for now until we implement implicit destinations.
```

Comment:
I copied this comment from another place because the justification is still the same.